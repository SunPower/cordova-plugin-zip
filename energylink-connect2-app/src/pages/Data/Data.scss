@import '~@sunpower/theme-dark/styles/variables';

.data {
  overflow-x: hidden;
  flex-direction: column;
  .section {
    margin: auto 0;
  }
  span.sp-map {
    font-size: 4rem;
  }

  .eqs-status {
    background-color: $space-dark;
    text-align: left;
    margin: 10px;
    padding: 10px;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
  }

  .no-meters-warning {
    background-color: $space-dark;
    border-radius: 10px;
    flex-direction: column;
  }
}
