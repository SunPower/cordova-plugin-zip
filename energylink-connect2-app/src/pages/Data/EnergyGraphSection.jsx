import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { filter, head, isNil, last, length, pathOr, propOr } from 'ramda'

import { useI18n } from 'shared/i18n'
import { DATA_SOURCES, GRAPHS, SELECT_ENERGY_GRAPH } from 'state/actions/user'
import EnergyGraph, { VIEWS } from 'components/EnergyGraph'
import EnergySwitch from 'components/EnergySwitch'

import './EnergyGraphSection.scss'
import { roundDecimals } from '../../shared/rounding'

const SelectedGraph = ({ selectedId, data, dataSource }) => {
  const t = useI18n()
  const inventory = useSelector(pathOr({}, ['inventory', 'bom']))
  const storageInventory = inventoryItem => inventoryItem.item === 'ESS'
  const storage = filter(storageInventory, inventory)
  const hasStorage = length(storage) ? head(storage).value !== '0' : false

  if (selectedId === GRAPHS.ENERGY) {
    return (
      <EnergyGraph
        className="mt-55"
        hasStorage={hasStorage}
        data={data}
        weather={true}
        view={VIEWS.LIVE}
        animation={false}
      />
    )
  }

  const dataEntries = Object.entries(data)
  const lastDataPoint =
    length(dataEntries) > 0
      ? last(dataEntries)[1]
      : {
          pp: 0,
          pc: 0,
          ps: 0
        }

  const produced = propOr(0, 'pp', lastDataPoint)
  const rawConsumed = roundDecimals(
    pathOr(0, ['rawData', 'site_load_p'], lastDataPoint)
  )
  const consumed = isNil(lastDataPoint.pc) ? rawConsumed : lastDataPoint.pc

  return (
    <React.Fragment>
      {dataSource === DATA_SOURCES.LIVE ? (
        <div className="graph-top-values mt-20">
          <div className="prod ml-10">
            <div>
              {produced}
              <span className="unit"> kW</span>
            </div>
            <div>{t('PRODUCED')}</div>
          </div>
          <div className="cons mr-10">
            <div>
              {consumed}
              <span className="unit"> kW</span>
            </div>
            <div>{t('CONSUMED')}</div>
          </div>
        </div>
      ) : (
        <div className="mt-55" />
      )}
      <EnergyGraph
        className="power-graph"
        series={['pp', 'pc', 'ps']}
        unitLabel="kW"
        hasStorage={hasStorage}
        power={true}
        data={data}
        weather={true}
        view={VIEWS.LIVE}
        animation={false}
      />
    </React.Fragment>
  )
}

const shareId = 'energy-graph'

export default function EnergyGraphSection() {
  const t = useI18n()
  const dispatch = useDispatch()

  const selectedId = useSelector(state => state.global.selectedEnergyGraph)
  const selectedDataSourceId = useSelector(
    state => state.global.selectedDataSource
  )

  const data = useSelector(state => state.energyLiveData.liveData)

  const graphSelect = [
    { id: GRAPHS.POWER, value: `${t('POWER')} (kW)` },
    { id: GRAPHS.ENERGY, value: `${t('ENERGY')} (kWh)` }
  ].map(entry =>
    entry.id === selectedId ? { ...entry, selected: true } : entry
  )

  return (
    <div
      id={shareId}
      className="energy-graph-container columns is-mobile is-multiline is-variable is-1"
    >
      <div className="column is-full ignore-in-share pb-0">
        <EnergySwitch
          entries={graphSelect}
          onChange={id => dispatch(SELECT_ENERGY_GRAPH(id))}
        />
      </div>
      <div className="column energy-graph-column is-full pt-0">
        <SelectedGraph
          selectedId={selectedId}
          data={data}
          dataSource={selectedDataSourceId}
        />
      </div>
    </div>
  )
}
