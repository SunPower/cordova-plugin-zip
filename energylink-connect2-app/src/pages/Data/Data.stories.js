import { storiesOf } from '@storybook/react'
import React from 'react'
import { Provider } from 'react-redux'

import Data from '.'

import { configureStore } from 'state/store'

const noMeters = {
  storage: {
    statusReport: {
      ess_report: {
        last_updated: '2020-02-15 01:23:45',
        battery_status: [
          {
            serial_number: 'string',
            last_updated: '2020-02-15 01:23:45',
            battery_amperage: {
              value: 0,
              unit: 'string'
            },
            battery_voltage: {
              value: 0,
              unit: 'string'
            },
            state_of_charge: {
              value: 0,
              unit: 'string'
            },
            temperature: {
              value: 0,
              unit: 'string'
            }
          }
        ],
        ess_status: [
          {
            last_updated: '2020-02-15 01:23:45',
            serial_number: 'string',
            enclosure_humidity: {
              value: 0,
              unit: 'string'
            },
            enclosure_temperature: {
              value: 0,
              unit: 'string'
            },
            ess_meter_reading: {
              last_updated: '2020-02-15 01:23:45',
              agg_power: {
                value: 0,
                unit: 'string'
              },
              meter_a: {
                reading: {
                  last_updated: '2020-02-15 01:23:45',
                  current: {
                    value: 0,
                    unit: 'string'
                  },
                  voltage: {
                    value: 0,
                    unit: 'string'
                  },
                  power: {
                    value: 0,
                    unit: 'string'
                  }
                }
              },
              meter_b: {
                reading: {
                  last_updated: '2020-02-15 01:23:45',
                  current: {
                    value: 0,
                    unit: 'string'
                  },
                  voltage: {
                    value: 0,
                    unit: 'string'
                  },
                  power: {
                    value: 0,
                    unit: 'string'
                  }
                }
              }
            }
          }
        ],
        hub_plus_status: {
          serial_number: 'string',
          last_updated: '2020-02-15 01:23:45',
          contactor_error: 'NONE',
          contactor_position: 'UNKNOWN',
          grid_voltage_state: 'METER_VOLTAGE_IN_RANGE',
          grid_frequency_state: 'METER_FREQ_IN_RANGE',
          load_voltage_state: 'METER_VOLTAGE_IN_RANGE',
          load_frequency_state: 'METER_FREQ_IN_RANGE',
          hub_temperature: {
            value: 0,
            unit: 'string'
          },
          hub_humidity: {
            value: 0,
            unit: 'string'
          },
          jump_start_voltage: {
            value: 0,
            unit: 'string'
          },
          aux_port_voltage: {
            value: 0,
            unit: 'string'
          },
          main_voltage: {
            value: 0,
            unit: 'string'
          },
          inverter_connection_voltage: {
            value: 0,
            unit: 'string'
          },
          grid_phase1_voltage: {
            value: 0,
            unit: 'string'
          },
          grid_phase2_voltage: {
            value: 0,
            unit: 'string'
          },
          load_phase1_voltage: {
            value: 0,
            unit: 'string'
          },
          load_phase2_voltage: {
            value: 0,
            unit: 'string'
          }
        },
        inverter_status: [
          {
            serial_number: 'string',
            last_updated: '2020-02-15 01:23:45',
            ac_current: {
              value: 0,
              unit: 'string'
            },
            phase_a_current: {
              value: 0,
              unit: 'string'
            },
            phase_b_current: {
              value: 0,
              unit: 'string'
            },
            a_n_voltage: {
              value: 0,
              unit: 'string'
            },
            b_n_voltage: {
              value: 0,
              unit: 'string'
            },
            ac_power: {
              value: 0,
              unit: 'string'
            },
            temperature: {
              value: 0,
              unit: 'string'
            }
          }
        ],
        ess_state: {
          storage_controller_status: 'UNKNOWN',
          operational_mode: 'UNKNOWN',
          permission_to_operate: true
        }
      },
      errors: [
        {
          error_name: 'UNDER_VOLT_ALARM',
          last_occurrence: '2020-02-15 01:23:45',
          error_code: '4.5.1',
          device_sn: '048572340857NND',
          error_message: 'Critical: low battery SOH.',
          value: {
            value: 0,
            unit: 'string'
          }
        }
      ]
    }
  },
  devices: {
    found: [
      {
        SERIAL: 'ProductionMeter001p',
        DEVICE_TYPE: 'Power Meter',
        SUBTYPE: 'NOT_USED'
      },
      {
        SERIAL: 'ConsumptionMeter001c',
        DEVICE_TYPE: 'Power Meter',
        SUBTYPE: 'NOT_USED'
      }
    ]
  },
  energyLiveData: {
    liveData: {
      [new Date().toISOString()]: {
        isSolarAvailable: false,
        ess_en: -162.80300000000005,
        ess_p: -0.041,
        net_en: -2157.16,
        net_p: -0.8628927917480469,
        pv_en: 2228.32,
        site_load_en: -91.64299999999974,
        site_load_p: -0.016651275634765615,
        soc: 0.9500000000000001,
        rawData: {
          ess_en: -162.80300000000005,
          ess_p: -0.041,
          net_en: -2157.16,
          net_p: -0.8628927917480469,
          pv_en: 2228.32,
          pv_p: 0.8872415161132813,
          site_load_en: -91.64299999999974,
          site_load_p: -0.016651275634765615,
          soc: 0.9500000000000001
        }
      }
    }
  }
}
const withMeters = {
  storage: {
    statusReportError: 'Error Here'
  },
  devices: {
    found: [
      {
        SERIAL: 'ProductionMeter001p',
        DEVICE_TYPE: 'Power Meter',
        SUBTYPE: 'GROSS_PRODUCTION_LINESIDE'
      },
      {
        SERIAL: 'ConsumptionMeter001c',
        DEVICE_TYPE: 'Power Meter',
        SUBTYPE: 'NOT_USED'
      }
    ]
  },
  energyLiveData: {
    liveData: {
      [new Date().toISOString()]: {
        isSolarAvailable: true,
        ess_en: -162.80300000000005,
        ess_p: -0.041,
        net_en: -2157.16,
        net_p: -0.8628927917480469,
        pv_en: 2228.32,
        pv_p: 0.8872415161132813,
        site_load_en: -91.64299999999974,
        site_load_p: -0.016651275634765615,
        soc: 0.9500000000000001,
        rawData: {
          ess_en: -162.80300000000005,
          ess_p: -0.041,
          net_en: -2157.16,
          net_p: -0.8628927917480469,
          pv_en: 2228.32,
          pv_p: 0.8872415161132813,
          site_load_en: -91.64299999999974,
          site_load_p: -0.016651275634765615,
          soc: 0.9500000000000001
        }
      }
    }
  }
}
const onlyMicroInverters = {
  storage: {
    statusReportError: 'Error Here'
  },
  devices: {
    found: [
      {
        SERIAL: 'ConsumptionMeter001ca',
        DEVICE_TYPE: 'Inverter',
        SUBTYPE: 'NOT_USED',
        MODEL: 'inverter'
      },
      {
        SERIAL: 'ConsumptionMeter002ca',
        DEVICE_TYPE: 'Inverter',
        SUBTYPE: 'NOT_USED',
        MODEL: 'inverter'
      }
    ]
  },
  energyLiveData: {
    liveData: {
      [new Date().toISOString()]: {
        isSolarAvailable: true,
        ess_en: -162.80300000000005,
        ess_p: -0.041,
        net_en: -2157.16,
        net_p: -0.8628927917480469,
        pv_en: 2228.32,
        pv_p: 0.8872415161132813,
        site_load_en: -91.64299999999974,
        site_load_p: -0.016651275634765615,
        soc: 0.9500000000000001,
        rawData: {
          ess_en: -162.80300000000005,
          ess_p: -0.041,
          net_en: -2157.16,
          net_p: -0.8628927917480469,
          pv_en: 2228.32,
          pv_p: 0.8872415161132813,
          site_load_en: -91.64299999999974,
          site_load_p: -0.016651275634765615,
          soc: 0.9500000000000001
        }
      }
    }
  }
}
const onlyStringInverters = {
  storage: {
    statusReportError: 'Error Here'
  },
  devices: {
    found: [
      {
        ISDETAIL: true,
        SERIAL: 'E00121943009545',
        TYPE: 'SOLARBRIDGE',
        STATE: 'working',
        STATEDESCR: 'Working',
        MODEL: 'AC_Module_Type_E',
        DESCR: 'Inverter E00121943009545',
        DEVICE_TYPE: 'Inverter',
        SWVER: '4.18.2',
        PORT: '',
        MOD_SN: '',
        NMPLT_SKU: '',
        DATATIME: '2020,11,04,19,28,59',
        ltea_3phsum_kwh: '258.6419',
        p_3phsum_kw: '0',
        vln_3phavg_v: '0',
        i_3phsum_a: '0',
        p_mppt1_kw: '0.001',
        v_mppt1_v: '63.12',
        i_mppt1_a: '0.01',
        t_htsnk_degc: '38',
        freq_hz: '67.98',
        stat_ind: '0',
        origin: 'data_logger',
        OPERATION: 'noop',
        CURTIME: '2020,11,04,19,29,15'
      },
      {
        ISDETAIL: true,
        SERIAL: 'E00121919007686',
        TYPE: 'SOLARBRIDGE',
        STATE: 'working',
        STATEDESCR: 'Working',
        MODEL: 'AC_Module_Type_E',
        DESCR: 'Inverter E00121919007686',
        DEVICE_TYPE: 'Inverter',
        PANEL: 'SPR-X20-327-BLK-E-AC',
        SWVER: '4.18.98',
        PORT: '',
        MOD_SN: '',
        NMPLT_SKU: '',
        DATATIME: '2020,11,04,19,28,59',
        ltea_3phsum_kwh: '403.9664',
        p_3phsum_kw: '0',
        vln_3phavg_v: '0',
        i_3phsum_a: '0',
        p_mppt1_kw: '0.0008',
        v_mppt1_v: '63.3',
        i_mppt1_a: '0.01',
        t_htsnk_degc: '38',
        freq_hz: '0',
        stat_ind: '0',
        origin: 'data_logger',
        OPERATION: 'noop',
        CURTIME: '2020,11,04,19,29,15'
      }
    ]
  },
  energyLiveData: {
    liveData: {
      [new Date().toISOString()]: {
        isSolarAvailable: true,
        ess_en: -162.80300000000005,
        ess_p: -0.041,
        net_en: -2157.16,
        net_p: -0.8628927917480469,
        pv_en: 2228.32,
        pv_p: 0.8872415161132813,
        site_load_en: -91.64299999999974,
        site_load_p: -0.016651275634765615,
        soc: 0.9500000000000001,
        rawData: {
          ess_en: -162.80300000000005,
          ess_p: -0.041,
          net_en: -2157.16,
          net_p: -0.8628927917480469,
          pv_en: 2228.32,
          pv_p: 0.8872415161132813,
          site_load_en: -91.64299999999974,
          site_load_p: -0.016651275634765615,
          soc: 0.9500000000000001
        }
      }
    }
  }
}

storiesOf('Data', module)
  .add('With Meters', () => {
    const { store } = configureStore(withMeters)

    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <Data />
        </Provider>
      </div>
    )
  })
  .add('Without Meter Configs', () => {
    const { store } = configureStore(noMeters)

    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <Data />
        </Provider>
      </div>
    )
  })
  .add('With only String Inverters', () => {
    const { store } = configureStore(onlyStringInverters)

    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <Data />
        </Provider>
      </div>
    )
  })
  .add('With only Micro Inverters', () => {
    const { store } = configureStore(onlyMicroInverters)

    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <Data />
        </Provider>
      </div>
    )
  })
