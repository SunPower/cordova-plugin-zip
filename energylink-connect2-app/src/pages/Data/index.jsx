import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  always,
  cond,
  equals,
  filter,
  head,
  length,
  pathOr,
  isEmpty,
  T
} from 'ramda'
import { useI18n } from 'shared/i18n'
import { roundDecimals } from 'shared/rounding'
import {
  ENERGY_DATA_START_POLLING,
  ENERGY_DATA_STOP_POLLING
} from 'state/actions/energy-data'
import { RUN_EQS_SYSTEMCHECK_SUCCESS } from 'state/actions/storage'
import { MI_DATA_START_POLLING, MI_DATA_STOP_POLLING } from 'state/actions/pvs'
import { either } from 'shared/utils'
import EnergyGraphSection from './EnergyGraphSection'
import RightNow from 'components/RightNow'
import MiDataLive from 'components/MiDataLive'
import Collapsible from 'components/Collapsible'
import { ButtonLink } from 'components/ButtonLink'
import paths from 'routes/paths'
import './Data.scss'

export default () => {
  const t = useI18n()
  const dispatch = useDispatch()
  const { liveData = {} } = useSelector(state => state.energyLiveData)
  const { miData } = useSelector(state => state.pvs)
  const inventory = useSelector(pathOr({}, ['inventory', 'bom']))
  const storageInventory = inventoryItem => inventoryItem.item === 'ESS'
  const storage = filter(storageInventory, inventory)
  const hasStorage = length(storage) ? head(storage).value !== '0' : false
  const storageStatus = useSelector(pathOr({}, ['storage', 'status']))
  const essState = pathOr({}, ['results', 'ess_report', 'ess_state', 0])(
    storageStatus
  )

  const essDisplayStatus = cond([
    [s => s.permission_to_operate, always('CONTROLLER RUNNING NORMALLY')],
    [
      s => equals('RUNNING', s.storage_controller_status),
      always('CONTROLLER IN PRE-PTO MODE')
    ],
    [
      s => equals('NOT_RUNNING', s.storage_controller_status),
      always('CONTROLLER NOT RUNNING')
    ],
    [T, always('CONTROLLER_DATA_ERROR')]
  ])

  useEffect(() => {
    dispatch(ENERGY_DATA_START_POLLING())
    dispatch(MI_DATA_START_POLLING())
    dispatch(RUN_EQS_SYSTEMCHECK_SUCCESS())
    return () => {
      dispatch(ENERGY_DATA_STOP_POLLING())
      dispatch(MI_DATA_STOP_POLLING())
    }
  }, [dispatch])

  let data = {
    solar: 0,
    stateOfCharge: 0,
    storage: 0,
    grid: 0,
    date: new Date(),
    homeUsage: 0
  }

  const entries = Object.entries(liveData)

  if (entries.length) {
    const [latestDate, latest] = entries[entries.length - 1]
    data = {
      date: latestDate,
      stateOfCharge: latest.soc,
      solar: latest.p,
      storage: latest.s,
      homeUsage: latest.c,
      grid: roundDecimals(latest.c - latest.p - latest.s),
      powerSolar: latest.pp,
      powerStorage: latest.ps,
      powerHomeUsage: latest.pc,
      powerGrid: roundDecimals(latest.pc - latest.pp - latest.ps)
    }
  }

  return (
    <section className="data is-flex has-text-centered full-height pl-10 pr-10">
      <section>
        {either(
          hasStorage,
          <Collapsible title={t('SUNVAULT_STATUS')}>
            <span className="has-text-weight-bold has-text-white mt-10 mb-10">
              {either(
                !isEmpty(essState),
                <span>
                  {storageStatus.error
                    ? t('EQS_ERROR')
                    : t(essDisplayStatus(essState))}
                </span>,
                <span className="loader-label">{t('LOADING')}</span>
              )}
            </span>
            <ButtonLink
              title={t('HEALTH_CHECK')}
              path={paths.PROTECTED.ESS_HEALTH_CHECK.path}
            />
          </Collapsible>
        )}
        {either(length(miData) > 0, <MiDataLive data={miData} />)}
      </section>
      <section>
        <h6 className="is-uppercase mt-20 mb-20">{t('RIGHT_NOW')}</h6>
        <RightNow
          solarValue={data.powerSolar}
          gridValue={data.powerGrid}
          hasStorage={hasStorage}
          storageValue={data.powerStorage}
          homeValue={data.powerHomeUsage}
          batteryLevel={data.stateOfCharge}
        />
      </section>
      <div className="separator" />
      <EnergyGraphSection />
    </section>
  )
}
