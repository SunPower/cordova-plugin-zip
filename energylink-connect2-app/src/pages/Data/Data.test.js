import React from 'react'
import Data from '.'
import { DATA_SOURCES } from 'state/actions/user'
import * as userActions from 'state/actions/user'
import * as i18n from 'shared/i18n'

describe('Data page', () => {
  const provider = {
    storage: {
      statusReport: {
        ess_report: {
          last_updated: '2020-02-15 01:23:45',
          battery_status: [
            {
              serial_number: 'string',
              last_updated: '2020-02-15 01:23:45',
              battery_amperage: {
                value: 0,
                unit: 'string'
              },
              battery_voltage: {
                value: 0,
                unit: 'string'
              },
              state_of_charge: {
                value: 0,
                unit: 'string'
              },
              temperature: {
                value: 0,
                unit: 'string'
              }
            }
          ],
          ess_status: [
            {
              last_updated: '2020-02-15 01:23:45',
              serial_number: 'string',
              enclosure_humidity: {
                value: 0,
                unit: 'string'
              },
              enclosure_temperature: {
                value: 0,
                unit: 'string'
              },
              ess_meter_reading: {
                last_updated: '2020-02-15 01:23:45',
                agg_power: {
                  value: 0,
                  unit: 'string'
                },
                meter_a: {
                  reading: {
                    last_updated: '2020-02-15 01:23:45',
                    current: {
                      value: 0,
                      unit: 'string'
                    },
                    voltage: {
                      value: 0,
                      unit: 'string'
                    },
                    power: {
                      value: 0,
                      unit: 'string'
                    }
                  }
                },
                meter_b: {
                  reading: {
                    last_updated: '2020-02-15 01:23:45',
                    current: {
                      value: 0,
                      unit: 'string'
                    },
                    voltage: {
                      value: 0,
                      unit: 'string'
                    },
                    power: {
                      value: 0,
                      unit: 'string'
                    }
                  }
                }
              }
            }
          ],
          hub_plus_status: {
            serial_number: 'string',
            last_updated: '2020-02-15 01:23:45',
            contactor_error: 'NONE',
            contactor_position: 'UNKNOWN',
            grid_voltage_state: 'METER_VOLTAGE_IN_RANGE',
            grid_frequency_state: 'METER_FREQ_IN_RANGE',
            load_voltage_state: 'METER_VOLTAGE_IN_RANGE',
            load_frequency_state: 'METER_FREQ_IN_RANGE',
            hub_temperature: {
              value: 0,
              unit: 'string'
            },
            hub_humidity: {
              value: 0,
              unit: 'string'
            },
            jump_start_voltage: {
              value: 0,
              unit: 'string'
            },
            aux_port_voltage: {
              value: 0,
              unit: 'string'
            },
            main_voltage: {
              value: 0,
              unit: 'string'
            },
            inverter_connection_voltage: {
              value: 0,
              unit: 'string'
            },
            grid_phase1_voltage: {
              value: 0,
              unit: 'string'
            },
            grid_phase2_voltage: {
              value: 0,
              unit: 'string'
            },
            load_phase1_voltage: {
              value: 0,
              unit: 'string'
            },
            load_phase2_voltage: {
              value: 0,
              unit: 'string'
            }
          },
          inverter_status: [
            {
              serial_number: 'string',
              last_updated: '2020-02-15 01:23:45',
              ac_current: {
                value: 0,
                unit: 'string'
              },
              phase_a_current: {
                value: 0,
                unit: 'string'
              },
              phase_b_current: {
                value: 0,
                unit: 'string'
              },
              a_n_voltage: {
                value: 0,
                unit: 'string'
              },
              b_n_voltage: {
                value: 0,
                unit: 'string'
              },
              ac_power: {
                value: 0,
                unit: 'string'
              },
              temperature: {
                value: 0,
                unit: 'string'
              }
            }
          ],
          ess_state: {
            storage_controller_status: 'UNKNOWN',
            operational_mode: 'UNKNOWN',
            permission_to_operate: true
          }
        },
        errors: [
          {
            error_name: 'UNDER_VOLT_ALARM',
            last_occurrence: '2020-02-15 01:23:45',
            error_code: '4.5.1',
            device_sn: '048572340857NND',
            error_message: 'Critical: low battery SOH.',
            value: {
              value: 0,
              unit: 'string'
            }
          }
        ]
      }
    },
    global: {
      selectedEnergyGraph: userActions.GRAPHS.ENERGY,
      selectedDataSource: DATA_SOURCES.LIVE
    },
    pvs: {
      miData: null
    },
    devices: {
      found: [
        {
          SERIAL: 'ProductionMeter001p',
          DEVICE_TYPE: 'Power Meter',
          SUBTYPE: 'GROSS_PRODUCTION_LINESIDE'
        },
        {
          SERIAL: 'ConsumptionMeter001c',
          DEVICE_TYPE: 'Power Meter',
          SUBTYPE: 'NOT_USED'
        }
      ]
    },
    energyLiveData: {
      liveData: {}
    }
  }

  beforeEach(() => {
    const baseTime = 1566947470000 // Aug 27 2019 23:11:10 GMT-0000
    jest.spyOn(Date, 'now').mockImplementation(() => baseTime)

    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key} ${params.join('_')}`.trim()
      )
  })

  test('render correctly', () => {
    const { component } = mountWithProvider(<Data />)(provider)
    expect(component).toMatchSnapshot()
  })
})
