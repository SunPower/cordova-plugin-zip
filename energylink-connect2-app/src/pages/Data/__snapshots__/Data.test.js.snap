// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data page render correctly 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <_default>
        <section
          className="data is-flex has-text-centered full-height pl-10 pr-10"
        >
          <section />
          <section>
            <div
              className="live-power-title pt-20 pb-20"
            >
              <div />
              <h6
                className="title-text is-uppercase"
              >
                LIVE_DATA
              </h6>
              <div
                className="info"
                onClick={[Function]}
              >
                <span
                  className="sp-info has-text-primary"
                />
              </div>
            </div>
            <div
              className="no-meters-warning has-text-weight-bold has-text-centered is-flex pt-20 pb-20"
            >
              <span
                className="has-text-white is-size-5"
              >
                LIVE_DATA_UNAVAILABLE
              </span>
              <span>
                NO_METER_CONFIG
              </span>
            </div>
          </section>
          <div
            className="separator"
          />
          <EnergyGraphSection>
            <div
              className="energy-graph-container columns is-mobile is-multiline is-variable is-1"
              id="energy-graph"
            >
              <div
                className="column is-full ignore-in-share pb-0"
              >
                <EnergySwitch
                  entries={
                    Array [
                      Object {
                        "id": "power",
                        "value": "POWER (kW)",
                      },
                      Object {
                        "id": "energy",
                        "selected": true,
                        "value": "ENERGY (kWh)",
                      },
                    ]
                  }
                  onChange={[Function]}
                >
                  <div
                    className="energy-switch is-flex"
                  >
                    <div
                      className="selected-label selected-1"
                    />
                    <div
                      className="entries is-flex pt-8 pb-8"
                    >
                      <div
                        className="power"
                        key="POWER (kW)"
                        onClick={[Function]}
                      >
                        POWER (kW)
                      </div>
                      <div
                        className="energy selected"
                        key="ENERGY (kWh)"
                        onClick={[Function]}
                      >
                        ENERGY (kWh)
                      </div>
                    </div>
                  </div>
                </EnergySwitch>
              </div>
              <div
                className="column energy-graph-column is-full pt-0"
              >
                <SelectedGraph
                  data={Object {}}
                  dataSource="LIVE"
                  selectedId="energy"
                >
                  <EnergyGraph
                    animation={false}
                    className="mt-55"
                    data={Object {}}
                    hasStorage={false}
                    view="live"
                    weather={true}
                  >
                    <div
                      className="energy-graph-container"
                    >
                      <ResponsiveContainer
                        className="energy-graph mt-55"
                        debounce={0}
                        height={380}
                        width="100%"
                      >
                        <div
                          className="recharts-responsive-container energy-graph mt-55"
                          style={
                            Object {
                              "height": 380,
                              "maxHeight": undefined,
                              "minHeight": undefined,
                              "minWidth": undefined,
                              "width": "100%",
                            }
                          }
                        >
                          <ComposedChart
                            barCategoryGap="10%"
                            barGap={4}
                            data={Array []}
                            height={380}
                            layout="horizontal"
                            margin={
                              Object {
                                "left": 5,
                                "right": 10,
                              }
                            }
                            reverseStackOrder={false}
                            stackOffset="none"
                            width={0}
                          />
                          <ResizeDetector
                            handleHeight={true}
                            handleWidth={true}
                            onResize={[Function]}
                            refreshRate={1000}
                            resizableElementId=""
                            skipOnMount={false}
                          >
                            <div
                              style={
                                Object {
                                  "display": "none",
                                  "height": 0,
                                  "position": "absolute",
                                  "visibility": "hidden",
                                  "width": 0,
                                }
                              }
                            />
                          </ResizeDetector>
                        </div>
                      </ResponsiveContainer>
                    </div>
                  </EnergyGraph>
                </SelectedGraph>
              </div>
            </div>
          </EnergyGraphSection>
          <SwipeableSheet
            onChange={[Function]}
            open={false}
          >
            <SwipeableBottomSheet
              defaultOpen={false}
              fullScreen={false}
              marginTop={0}
              onChange={[Function]}
              open={false}
              overflowHeight={0}
              overlay={true}
              scrollTopAtClose={true}
              shadowTip={true}
              swipeableViewsProps={
                Object {
                  "className": "swipeable",
                }
              }
              topShadow={true}
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "height": 0,
                    "left": 0,
                    "position": "fixed",
                    "right": 0,
                  }
                }
              >
                <HeightUpdater
                  height={768}
                  onHeightChange={[Function]}
                />
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "black",
                      "height": 768,
                      "left": 0,
                      "opacity": 0,
                      "pointerEvents": "none",
                      "position": "fixed",
                      "right": 0,
                      "top": 0,
                      "transition": "opacity 450ms",
                    }
                  }
                />
                <ReactSwipableView
                  animateHeight={false}
                  animateTransitions={true}
                  axis="y"
                  className="swipeable"
                  containerStyle={
                    Object {
                      "boxSizing": "border-box",
                    }
                  }
                  disableLazyLoading={false}
                  disabled={false}
                  enableMouseEvents={true}
                  hysteresis={0.6}
                  ignoreNativeScroll={false}
                  index={0}
                  onChangeIndex={[Function]}
                  onTransitionEnd={[Function]}
                  resistance={false}
                  slideStyle={
                    Object {
                      "boxSizing": "border-box",
                      "marginBottom": -0,
                      "overflow": "visible",
                    }
                  }
                  springConfig={
                    Object {
                      "delay": "0s",
                      "duration": "0.35s",
                      "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                    }
                  }
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "overflowY": "initial",
                    }
                  }
                  threshold={5}
                >
                  <div
                    className="swipeable"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onScroll={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "overflowY": "initial",
                      }
                    }
                  >
                    <div
                      className="react-swipeable-view-container"
                      style={
                        Object {
                          "WebkitFlexDirection": "column",
                          "WebkitTransition": "all 0s ease 0s",
                          "boxSizing": "border-box",
                          "direction": "ltr",
                          "display": "flex",
                          "flexDirection": "column",
                          "height": null,
                          "transition": "all 0s ease 0s",
                          "willChange": "transform",
                        }
                      }
                    >
                      <div
                        aria-hidden={false}
                        data-swipeable="true"
                        key=".0"
                        style={
                          Object {
                            "WebkitFlexShrink": 0,
                            "boxSizing": "border-box",
                            "flexShrink": 0,
                            "marginBottom": -0,
                            "overflow": "visible",
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="ReactSwipeableBottomSheet--closed"
                          style={
                            Object {
                              "backgroundColor": "white",
                              "height": "initial",
                              "maxHeight": 768,
                              "overflow": "hidden",
                            }
                          }
                        >
                          <div
                            className="live-power-info is-flex has-text-white"
                          >
                            <span
                              className="sp-info is-size-1"
                            />
                            <span
                              className="mt-10 mb-10 has-text-justified"
                            >
                              LIVE_POWER_INFO
                            </span>
                            <div
                              className="mt-10 mb-20"
                            >
                              <button
                                className="button is-primary"
                                onClick={[Function]}
                              >
                                CLOSE
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </ReactSwipableView>
              </div>
            </SwipeableBottomSheet>
          </SwipeableSheet>
        </section>
      </_default>
    </Provider>
  </Router>
</BrowserRouter>
`;
